ambry-tools/src/main/java/com.github.ambry/store/DumpIndexTool.java

```diff
@@ -42,481 +49,358 @@
 /**
  * Tool to assist in dumping indices in Ambry
  * Supported operations are
- * 1. Dump Index
- * 2. Dump Index for a replica
- * 3. Dump active blobs for an index
- * 4. Dump active blobs for a replica
- * 5. Dump N random blobs for a replica
+ * 1. Dump index.
+ * 2. Dump an index segment.
+ * 3. Verify an index for sanity (no duplicate records).
  */
 public class DumpIndexTool {
-  private final ClusterMap clusterMap;
-  // set to true if only error logging is required
-  private final boolean silent;
+
+  public class Info {
+    private final EnumSet<BlobState> states;
+    private final boolean isInRecentIndexSegment;
+
+    public Info(EnumSet<BlobState> states, boolean isInRecentIndexSegment) {
+      this.states = states;
+      this.isInRecentIndexSegment = isInRecentIndexSegment;
+    }
+
+    public EnumSet<BlobState> getStates() {
+      return states;
+    }
+
+    public boolean isInRecentIndexSegment() {
+      return isInRecentIndexSegment;
+    }
+  }
+
+  public class IndexProcessingResults {
+    private final Map<StoreKey, Info> keyToState;
+    private final long processedCount;
+    private final long putCount;
+    private final long deleteCount;
+    private final Set<StoreKey> duplicatePuts;
+    private final Set<StoreKey> putAfterDeletes;
+    private final Set<StoreKey> duplicateDeletes;
+    private final long activeCount;
+
+    public IndexProcessingResults(Map<StoreKey, Info> keyToState, long processedCount, long putCount, long deleteCount,
+        Set<StoreKey> duplicatePuts, Set<StoreKey> putAfterDeletes, Set<StoreKey> duplicateDeletes) {
+      this.keyToState = keyToState;
+      this.processedCount = processedCount;
+      this.putCount = putCount;
+      this.deleteCount = deleteCount;
+      this.duplicatePuts = duplicatePuts;
+      this.putAfterDeletes = putAfterDeletes;
+      this.duplicateDeletes = duplicateDeletes;
+      activeCount = keyToState.values().stream().filter(value -> value.getStates().contains(BlobState.Valid)).count();
+    }
+
+    public Map<StoreKey, Info> getKeyToState() {
```

Comment:
who are the consumers of these getters?

Reply:
`getKeyToState()` is used by the `ConsistencyCheckerTool`. As for the others, none as of now but this is meant to be an API and I think users of the API might have use for this.

Also, this is replicating something very close to what `IndexStats` used to be.

Reply:
I would like to follow some standards across the board. I have been questioned in some of my patches, as to why we have some additions methods or apis for which we don't have any consumers for now and the request was that unless we have a requirement, lets not add it. Can we follow the same, or have some standard practices as to when such methods/apis are allowed so that all of us are in same page.